libcloudforensics/gcp_test.py

```diff
@@ -16,28 +16,343 @@
 
 from __future__ import unicode_literals
 
+import binascii
 import unittest
 
+from googleapiclient.errors import HttpError
+
 from libcloudforensics import gcp
 
 import mock
 import six
 
+FAKE_ANALYSIS_PROJECT = gcp.GoogleCloudProject(
+  'test-target-project',
+  'fake-zone'
+)
+FAKE_ANALYSIS_VM = gcp.GoogleComputeInstance(
+  FAKE_ANALYSIS_PROJECT,
+  'fake-zone',
+  'fake-analysis-vm'
+)
+FAKE_INSTANCE = gcp.GoogleComputeInstance(
+  FAKE_ANALYSIS_PROJECT,
+  'fake-zone',
+  'fake-instance'
+)
+FAKE_DISK = gcp.GoogleComputeDisk(
+  FAKE_ANALYSIS_PROJECT,
+  'fake-zone',
+  'disk1'
+)
+FAKE_BOOT_DISK = gcp.GoogleComputeDisk(
+  FAKE_ANALYSIS_PROJECT,
+  'fake-zone-other',
+  'boot-disk'
+)
+FAKE_SNAPSHOT = gcp.GoogleComputeSnapshot(
+  FAKE_DISK,
+  'snapshot1'
+)
+FAKE_DISK_COPY = gcp.GoogleComputeDisk(
+  FAKE_ANALYSIS_PROJECT,
+  'fake-zone',
+  'disk1-copy'
+)
+
 
 class GoogleCloudProjectTest(unittest.TestCase):
   """Test Google Cloud project class."""
 
   def setUp(self):
     super(GoogleCloudProjectTest, self).setUp()
-    self.project = gcp.GoogleCloudProject('test-project')
-    self.project.GceApi = mock.MagicMock()
-    self.project.GceOperation = mock.MagicMock()
 
-  def testFormatLogMessage(self):
-    """Test formatting log message."""
+  def test_format_log_message(self):
+    """Test formatting log message. """
     msg = 'Test message'
-    formatted_msg = self.project.format_log_message(msg)
+    formatted_msg = FAKE_ANALYSIS_PROJECT.format_log_message(msg)
     self.assertIsInstance(formatted_msg, six.string_types)
     self.assertEqual(
-        formatted_msg, u'project:{0:s} {1:s}'.format(
-            self.project.project_id, msg))
+      formatted_msg, u'project:{0:s} {1:s}'.format(
+        FAKE_ANALYSIS_PROJECT.project_id, msg))
+
+  @mock.patch('libcloudforensics.gcp.GoogleCloudProject.gce_operation')
+  @mock.patch('libcloudforensics.gcp.GoogleCloudProject.gce_api')
+  def test_list_instances(self, mock_gce_api, mock_gce_operation):
+    """Test that instances of project are correctly listed. """
+    mock_gce_api.return_value.instances.return_value.aggregatedList.return_value.execute.return_value = None
+    mock_gce_operation.return_value = {
+      # See https://cloud.google.com/compute/docs/reference/rest/v1/instances/aggregatedList
+      # for complete structure
+      'items': {
+        0: {
+          'instances': [
+            {
+              'name': FAKE_INSTANCE.name,
+              'zone': '/' + FAKE_INSTANCE.zone
+            }
+          ]
+        }
+      }
+    }
```

Comment:
As discussed, maybe set these long dicts at the top of the file for better test readability.

Reply:
Addressed